internal/controller/execplan.go

```diff
@@ -109,19 +109,24 @@ func (r *TemporalWorkerDeploymentReconciler) executePlan(ctx context.Context, l
 			}); err != nil { // would be cool to do this atomically with the update
 				return fmt.Errorf("unable to update metadata after setting current deployment: %w", err)
 			}
-		} else if ramp := vcfg.rampPercentage; ramp > 0 { // TODO(carlydf): Support setting any ramp in [0,100]
+		} else if ramp := vcfg.rampPercentage; ramp > 0 || vcfg.unsetRamp { // TODO(carlydf): Support setting any ramp in [0,100]
 			err := awaitVersionRegistration(ctx, l, deploymentHandler, p.TemporalNamespace, vcfg.versionID)
 			if err != nil {
 				return fmt.Errorf("error waiting for version to register, did your pollers start successfully?: %w", err)
 			}
 
-			l.Info("applying ramp", "version", p.UpdateVersionConfig.versionID, "percentage", p.UpdateVersionConfig.rampPercentage)
+			desiredRampingVersion := vcfg.versionID
+			if vcfg.unsetRamp {
+				desiredRampingVersion = ""
+			}
+
+			l.Info("applying ramp", "version", desiredRampingVersion, "percentage", p.UpdateVersionConfig.rampPercentage)
```

Comment:
I think this log could potentially be confusing; it won't be possible to disambiguate between a ramp being "removed" vs. modified as expected during a normal progressive rollout based on the message, and the `version` tag being empty isn't very intuitive. I'd like to emit a different log message when ramp is being removed.

Maybe moving the code responsible for removing ramp into a separate if statement would help? Hardcoding request values of `Version: "", Percentage: 0` would also make the API call more clear IMO (I'm not sure right now if the value of `vcfg.rampPercentage` matters as long as `desiredRampingVersion` is empty).

Reply:
right now, the API requires a 0% ramp to be passed if "unset" is happening. But we are getting rid of that with the struct change because it's confusing